2014_TPDS_basic_decomposition_based_algorithm/src/decomposition.rs

```diff
@@ -10,13 +10,14 @@ pub fn decompose(dag: &mut Graph<NodeData, i32>) {
     let mut segments = create_segments(dag);
     calculate_segments_deadline(dag, &mut segments);
 
+    // Add deadlines to nodes.
     let mut nodes_deadline = vec![0.0; dag.node_count()];
     for segment in segments.iter() {
         segment.nodes.iter().for_each(|node| {
             nodes_deadline[node.id as usize] += segment.deadline;
         });
     }
-
+    // Convert to rational numbers to handle floating point numbers.
```

Comment:
It is more efficient to include this process in the previous loop.

Reply:
Since node deadlines cannot be obtained until all segment deadlines have been calculated, they cannot be included in the previous loop.
Converting segment.deadline to an integer and then inserting it is fine, but the loop count will be higher.